lib/applicants/field-resolvers.ts

```diff
@@ -160,3 +160,35 @@ export const applicantFileHistoryResolver: Resolver<Applicant> = async (
 
   return applicationProcessings;
 };
+
+/**
+ * Field resolver to fetch the most recent renewal request of an applicant.
+ * If the applicant has no previous renewals, return the most recent application.
+ * @returns Application object
+ */
+export const applicantMostRecentRenewalApplicationResolver: Resolver<Applicant> = async (
+  parent,
+  _args,
+  { prisma }
+) => {
+  const renewal = await prisma.applicant
+    .findUnique({
+      where: { id: parent.id },
+    })
+    .applications({
+      orderBy: [
+        {
+          isRenewal: SortOrder.DESC,
+        },
```

Comment:
The reason I have it within the sorting is so that if there are no previous renewals for an applicant, it will return the most recent replacement application. This is documented in the comments above the field resolver. Is that okay?

Reply:
Sounds good, that's a good approach!